### PR TITLE
Deprecate /cdns/configs

### LINKED ISSUE #3849: Rewrite /cdns/configs to Go
## I'm submitting a ...
- [ ] bug report
- [ ] new feature / enhancement request
- [x] improvement request (usability, performance, tech debt, etc.)
- [ ] other

## Traffic Control components affected ...
- [ ] CDN in a Box
- [x] Documentation (possibly)
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:
`/cdns/configs` is handled by Perl

## Expected / new behavior:
`/cdns/configs` should be handled by Go.

### Minimum API Version
1.2 - **This is the actual problem here**. The current Go implementation has a minimum supported version of 1.1

### Request Methods
- [ ] `GET`

## Minimal reproduction of the problem with instructions:
N/A

### PR DESCRIPTION
**Pull request by mhoppa**
## What does this PR (Pull Request) do?

- [x] This PR fixes #3849 

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

Make a request to `/cdns/configs` and make sure deprecation notice is returned

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: docs/source/api/cdns_configs.rst [hunk 1]
```diff
@@ -34,9 +35,37 @@ No parameters available.
...
+			"id": 2,
+			"name": "CDN-in-a-Box"
+		}
+	],
+    "alerts": [
+        {
+            "text": "This endpoint is deprecated, please use GET /cdns instead",
+            "level": "warning"
+        }
+    ]}
```
Comment (ocket8888) [inline docs/source/api/cdns_configs.rst]: Space-based indentation from `"alerts": [` down - pls use the master race indentation (at least to indent the code block - the JSON's indentation within that can thereon be spaces, although I think that's needlessly complicated).

File: traffic_ops/traffic_ops_golang/cdn/configs.go [hunk 2]
```diff
@@ -20,37 +20,33 @@ package cdn
...
 
-func GetConfigs(w http.ResponseWriter, r *http.Request) {
-	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
-	if userErr != nil || sysErr != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
-		return
-	}
-	defer inf.Close()
-	api.RespWriter(w, r, inf.Tx.Tx)(getConfigs(inf.Tx.Tx))
+//we need a type alias to define functions on
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/cdn/configs.go]: nit: GoDocs should be complete sentences starting with the name of the object they describe.

File: docs/source/api/cdns_configs.rst (hunk 1 above)
Comment (mhoppa) [reply docs/source/api/cdns_configs.rst]: man manual formatting documentation for API at this level is so incredibly painful, will fix.

Comment (ocket8888) [reply docs/source/api/cdns_configs.rst]: Just always use tabs for indentation everywhere and allow the peace and serenity they provide to wash over you, my friend.

Comment (mhoppa) [reply docs/source/api/cdns_configs.rst]: :)

**Pull request merged**